File: java_projects/glue_code/commons-codec/src/main/java/org/apache/commons/codec/language/bm/RuleType.java
package org.apache.commons.codec.language.bm;

import org.graalvm.polyglot.Value;
import org.graalvm.polyglot.PolyglotException;
import org.apache.commons.codec.ContextInitializer;
import org.apache.commons.codec.ExceptionHandler;
import org.apache.commons.codec.IntegrationUtils;

public enum RuleType {
    APPROX("approx"),
    EXACT("exact"),
    RULES("rules");

    private Value clz = ContextInitializer.getPythonClass("/language/bm/RuleType.py", "RuleType");
    private Value obj;

    RuleType(Value obj) {
        this.obj = obj;
    }

    public Value getPythonObject() {
        return obj;
    }

    public String getName() {
        //
        // return this.name;
        //

        // TODO: Check the type mapping below!
        return obj.invokeMember("getName").as(String.class);
    }

    RuleType(final String name) {
        //
        // this.name = name;
        //

        this.obj = clz.invokeMember("__init__", name);
    }
}

File: data/recomposed_projects/deepseek-coder-33b-instruct/body/commons-codec/src/main/org/apache/commons/codec/language/bm/RuleType.py
from __future__ import annotations
import re
import io
import typing
from typing import *


class RuleType:

    RULES: RuleType = None

    EXACT: RuleType = None

    APPROX: RuleType = None

    __name: str = ""

    def getName(self) -> str:
        return self.__name

    def __init__(self, name: str) -> None:

        self.__name = name

        if self.__name == "rules":
            RuleType.RULES = self
        elif self.__name == "exact":
            RuleType.EXACT = self
        elif self.__name == "approx":
            RuleType.APPROX = self
